modin/backends/base/query_compiler.py

```diff
@@ -1740,218 +3577,1371 @@ def set_index_names(self, names, axis=0):
+
+    @_doc_str_method(
+        method="index",
+        params="""sub : str
+        start : int, default: 0
+        end : int, optional""",
+    )
+    def str_index(self, sub, start=0, end=None):
+        return StrDefault.register(pandas.Series.str.index)(self, sub, start, end)
+
+    @_doc_str_method(method="isalnum", params="")
+    def str_isalnum(self):
+        return StrDefault.register(pandas.Series.str.isalnum)(self)
+
+    @_doc_str_method(method="isalpha", params="")
+    def str_isalpha(self):
+        return StrDefault.register(pandas.Series.str.isalpha)(self)
+
+    @_doc_str_method(method="isdecimal", params="")
+    def str_isdecimal(self):
+        return StrDefault.register(pandas.Series.str.isdecimal)(self)
+
+    @_doc_str_method(method="isdigit", params="")
+    def str_isdigit(self):
+        return StrDefault.register(pandas.Series.str.isdigit)(self)
+
+    @_doc_str_method(method="islower", params="")
+    def str_islower(self):
+        return StrDefault.register(pandas.Series.str.islower)(self)
+
+    @_doc_str_method(method="isnumeric", params="")
+    def str_isnumeric(self):
+        return StrDefault.register(pandas.Series.str.isnumeric)(self)
+
+    @_doc_str_method(method="isspace", params="")
+    def str_isspace(self):
+        return StrDefault.register(pandas.Series.str.isspace)(self)
+
+    @_doc_str_method(method="istitle", params="")
+    def str_istitle(self):
+        return StrDefault.register(pandas.Series.str.istitle)(self)
+
+    @_doc_str_method(method="isupper", params="")
+    def str_isupper(self):
+        return StrDefault.register(pandas.Series.str.isupper)(self)
+
+    @_doc_str_method(method="join", params="sep : str")
+    def str_join(self, sep):
+        return StrDefault.register(pandas.Series.str.join)(self, sep)
+
+    @_doc_str_method(method="len", params="")
+    def str_len(self):
+        return StrDefault.register(pandas.Series.str.len)(self)
+
+    @_doc_str_method(
+        method="ljust",
+        params="""width : int
+        fillchar : str, default: ' '""",
+    )
+    def str_ljust(self, width, fillchar=" "):
+        return StrDefault.register(pandas.Series.str.ljust)(self, width, fillchar)
+
+    @_doc_str_method(method="lower", params="")
+    def str_lower(self):
+        return StrDefault.register(pandas.Series.str.lower)(self)
+
+    @_doc_str_method(method="lstrip", params="to_strip : str, optional")
+    def str_lstrip(self, to_strip=None):
+        return StrDefault.register(pandas.Series.str.lstrip)(self, to_strip)
+
+    @_doc_str_method(
+        method="match",
+        params="""pat : str
+        case : bool, default: True
+        flags : int, default: 0
+        na : object, default: np.NaN""",
+    )
+    def str_match(self, pat, case=True, flags=0, na=np.NaN):
+        return StrDefault.register(pandas.Series.str.match)(self, pat, case, flags, na)
+
+    @_doc_str_method(method="normalize", params="form : {'NFC', 'NFKC', 'NFD', 'NFKD'}")
```

Comment:
uh, `form` parameter could use a description of what these cryptic acronyms mean...

Reply:
doc-string has a link to the pandas docs that explains these scary acronyms